### PR TITLE
Queues - Kaitlin Ramirez - NomNomKitchen

### PR DESCRIPTION
**Pull request by arrozconcurry**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | Postman, and saving result pages. Looking at the keys and values in the hashes. |
| Describe your API Wrapper.  How did you decide on the methods you created? | This was kinda trial and error. I started just trying everything in order to get pagination to work. I feel like I am calling things I don't need, like initialize in `lib/Edamam.rb` but without it it doesn't work.  |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I only have one right now. Than an empty search or search item with no results returns a blank page. I would like to come back and add more tests and safe guards in the wrapper for bad searches. |
| Explain how VCR aids in testing an API. | Can test the API without increasing calls. Save money or your allotment. |
| What is the Heroku URL of your deployed application? | http://nomnomkitchen.herokuapp.com/ |
| Do you have any recommendations on how we could improve this project for the next cohort? | Not that I can think of right now. I did a lot of googling to get the pagination to work and ended up using a gem. I left my attempted self pagination method in the controller. (Sidenote: I feel like I have more commits. I created different branches. I don't know if that affects the total count) |

### CODE REVIEW

File: lib/edamam.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def initialize(search_item, uri)
+    @search_item = search_item
+    @uri = fix_uri
+  end
+
+  def self.all_search(search_item)
+    query_params = {
+      "q" => search_item,
+      "id" => ENV["edamam_id"],
+      "key" => ENV["edamam_key"],
```
Comment (droberts-sea) [inline lib/edamam.rb:17]: Not a big deal, but typically environment variables are named using all caps, so these would be `EDAMAM_ID` and `EDAMAM_KEY`. Think of it like a Ruby constant.

File: app/views/searches/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <%= text_field_tag :search_item, params[:search_item], placeholder: "Search Recipes" %>
+    </div>
+  <div class="button small small-2 columns"> <%= submit_tag "Search", :name => nil %></div>
+  </div>
+<% end %>
+
+
+<% if @page_results %>
+  <section class="row small-up-1 medium-up-2 large-up-5">
+    <% @page_results.each do |result| %>
```
Comment (droberts-sea) [inline app/views/searches/index.html.erb:13]: While this certainly works, I don't know that I'm a big fan of reusing the search results view for the landing page. I would argue these are separate functions and ought to have separate routes and controller actions.

File: lib/edamam.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+class Edamam
+  class Edamam < StandardError; end
```
Comment (droberts-sea) [inline lib/edamam.rb:2]: Doesn't look like you actually use this error anywhere.

File: lib/edamam.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def self.show_one_result(uri)
+    query_params = {
+      "r" => uri,
+      "id" => ENV["edamam_id"],
+      "key" => ENV["edamam_key"]
+    }
+    response = HTTParty.get(BASE_URL, query: query_params).parsed_response
+    return response
```
Comment (droberts-sea) [inline lib/edamam.rb:50]: It's interesting to me that above you return `Result` objects, but here you return a hash. Seems to me you could add a little bit of functionality to `Result` and be able to reuse it nicely.

**Comment by droberts-sea** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | yes |
| Comprehension questions	| yes |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | yes |
| Semantic HTML | yes |
| Errors are reported to the user | no - setting up something for this (maybe a section in `application.html.erb` to display flash messages) should be one of your first steps for a new project, even if you aren't sure you're going to need it. |
| API Wrapper to handle the API requests | yes - see inline comments |
| Controller testing | good positive tests, but you're missing any kind of negative testing. What happens when the user inputs a search term that returns no results, or asks to show a recipe that doesn't exist? |
| **Styling** |  |
| Foundation Styling for responsive layout | yes |
| Search View | yes |
| List View | yes |
| Show View which opens in a new tab | yes (not in a new tab but looks good) |
| **API Features** | |
| The App attributes Edaman | yes |
| The VCR casettes do not contain the API key | yes |
|  **Overall** |  |

Good work overall! Be sure to see the inline comments, mostly small style things that don't really fit here.
